### PR TITLE
modeld: eliminate the memory copy to GPU 

### PR DESCRIPTION
**Pull request by pd0wm**
Continuation of https://github.com/commaai/openpilot/pull/2197

### CODE REVIEW

**Comment by pd0wm** [comment]
Doesn't pass the model replay test. Need to check why the model outputs change.

File: selfdrive/modeld/modeld.cc [hunk 1]
```diff
@@ -210,7 +212,9 @@ int main(int argc, char **argv) {
       }
 
     }
-    visionbuf_free(&yuv_ion);
+    for(auto &v : cl_buffers) {
+      clReleaseMemObject(v.second);
```
Comment (mitchellgoffpc) [inline selfdrive/modeld/modeld.cc:216]: is clReleaseMemObject all we need here? visionbuf_free also calls munmap and closes the buffer's file descriptor

**Comment by mitchellgoffpc** [comment]
> Doesn't pass the model replay test. Need to check why the model outputs change.

**Comment by geohot** [comment]
The reason the model output change I believe is because the test is broken.

If using the ION implementation for visionbuf_allocate_cl, the memcpy is okay since that's allocated with the UNCACHED flag. But it doesn't work for the CL implementation.

So I believe we've been testing with 0s the whole time.

File: selfdrive/modeld/modeld.cc (hunk 1 above)
Comment (pd0wm) [reply selfdrive/modeld/modeld.cc:216]: On the receiving end we don't have a fd to the ion buffer handle to close

**Comment by pd0wm** [comment]
I made another PR to add the sync: https://github.com/commaai/openpilot/pull/2675. Let's see if that also breaks the tests.

**Pull request closed without merging**